src/main/scala/com/fulcrumgenomics/util/Metric.scala

```diff
@@ -99,56 +99,74 @@ object Metric extends LazyLogging {
     }
   }
 
+  private[util] def build[T <: Metric](reflectiveBuilder: ReflectiveBuilder[T],
```

Comment:
next two methods refactored out of `def iterator` so `MetricSorterCodec` can use them